front/src/components/ThreeDModel/GLBModel.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { memo } from 'react'
+import { useGLTF } from '@react-three/drei'
+import * as THREE from 'three'
+
+export const GLBModel = memo(function GLBModel({
+  modelUrl,
+  modelPosition = [0, 0, 0],
+  modelScale = 1,
+  modelRotation = [0, 0, 0],
+  overrideColorHex,
+  shouldCastShadow = true,
+  shouldReceiveShadow = true,
+  forceFlatColorHex,
+  useLambertWhiteMaterial,
+}: {
+  modelUrl: string
+  modelPosition?: [number, number, number]
+  modelScale?: number | [number, number, number]
+  modelRotation?: [number, number, number]
+  overrideColorHex?: string
+  shouldCastShadow?: boolean
+  shouldReceiveShadow?: boolean
+  forceFlatColorHex?: string
+  useLambertWhiteMaterial?: boolean
+}) {
+  const { scene: gltfScene } = useGLTF(modelUrl)
+
+  React.useEffect(() => {
+    if (!gltfScene) return
+
+    gltfScene.traverse((object3D) => {
+      if ((object3D as THREE.Mesh).isMesh) {
```

Comment:
try to use `as` as much as possible
```suggestion
      if (object3D instanceof THREE.Mesh) {
```